### PR TITLE
Update binaryen

### PR DESCRIPTION
**Pull request by lqd**
it now has the more explicit operators @kripken mentioned
(not exactly master but close to it, for some temporary and minor VS2015 related things in binaryen)

### CODE REVIEW

File: src/trans.rs [hunk 1]
```diff
@@ -314,12 +314,12 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                     let a = self.trans_operand(a);
                     let b = self.trans_operand(b);
                     let op = match *op {
-                        BinOp::Add => BinaryenAdd(),
-                        BinOp::Sub => BinaryenSub(),
-                        BinOp::Mul => BinaryenMul(),
-                        BinOp::Div => BinaryenDivS(),
-                        BinOp::Eq => BinaryenEq(),
-                        BinOp::Ne => BinaryenNe(),
+                        BinOp::Add => BinaryenAddInt32(),
```
Comment (kripken) [inline src/trans.rs:317]: eventually there should be a check on the type, so we can handle the other types (i64, f32, f64). meanwhile maybe we could just assert that the type is i32? although, the binaryen validator should show an error if we get something like this wrong, so i guess it's ok as is too.

Comment (lqd) [inline src/trans.rs:317]: Yeah I agree, and this is why I was asking the question about the original behavior I wanted to preserve. I also saw that trans_operand panics if numerical constants were not i32 and those made me think this was ok for now but should definitely be dealt with as more of the numeric types are handled. assert + comment mentioning the TODO could be worthwhile right ?

Comment (kripken) [inline src/trans.rs:317]: Oh, I didn't notice the panic on non-i32. Sounds fine for now then, I think. A `TODO` sounds good though, would make it clear what next steps are.

**Comment by brson** [comment]
Thanks @lqd!

**Pull request merged**